### PR TITLE
added implementation for the task:

### PR DESCRIPTION
**Pull request by nikolasdymchenko7**
- added constants
- initialization elementData array in the ArrayList constructor
- additional methods: checkIndex(), findIndex(), copyToTempArray(), copyBackToArray(), addElement(), grow(), override method grow(), newCapacity()
- override methods: add(), addAll(), get(), set(), remove(), size(), isEmpty()

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,48 +1,170 @@
...
+            elementData[foundIndex] = value;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = null;
+        oldValue = (T) elementData[index];
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/ArrayList.java:83]: 2 lines?

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,48 +1,170 @@
...
+        if (index == size - FIRST_INDEX) {
+            size--;
+        } else {
+            Object[] tempArray = new Object[size - index];
+            index++;
+            tempArray = copyToTempArray(index, tempArray, ZERO_INDEX, tempArray.length);
+            index--;
+            size--;
+            elementData = copyBackToArray(index, tempArray, tempArray.length);
+        }
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/ArrayList.java:93]:
```suggestion
        System.arraycopy(elementData, index + 1, elementData, index, size - index);
        elementData[--size] = null;
```

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,48 +1,170 @@
...
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(elementData[i], element)) {
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/ArrayList.java:100]: image

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**